Ap weapons->develop (m4, pistol, grenades) also deleted the last of unitychan

ndru-pi-squared opened the pull request.

ndru-pi-squared committed:
Added easy weapons package and swapped the models on our guns. Also made the particle effect slightly more realistic.

Needs work

tua58997 committed:
Implemented friendly fire option

Default = no friendly fire. Set in Game Manager in inspector.
PlayerManager.TakeDamage(int,PlayerManager) checks if GameManager.FriendlyFire and if player taking damage is on the same team as player giving damage.

ndru-pi-squared committed:
minor changes to muzzleflash of pistol

ndru-pi-squared committed:
Replaced the RoombaCar model with a grenade model.

You can now press C to slide grenades across the floor with poor physics (no explosion yet)

ndru-pi-squared committed:
more minor changes: changed "RoombaCar" to "FragGrenade" in PlayerManager and in the hierarchy

ndru-pi-squared committed:
Added an IShootable interface along with a few others, serving as a framework to keep things organized as we build lots of items

ndru-pi-squared committed:
Merge branch 'develop' into AP-weapons

molongjoe reviewed (approved):
Thank you for deleting her and I love you also

ndru-pi-squared committed:
Separated out RoombaCar and FragGrenade.

Initially I had build the frag from the car and so the grenade model was essentially replacing the car model. But now we have both to use, along with the simple AI car script which may or may not be needed anymore. Currently working on the frag script
